brion/plugin/morphologyHDF5.cpp

```diff
@@ -352,6 +372,41 @@ Vector2isPtr MorphologyHDF5::readApicals() const
     return data;
 }
 
+floatsPtr MorphologyHDF5::readPerimeters() const
+{
+    if( _version != MORPHOLOGY_VERSION_H5_1_1 )
+        return floatsPtr( new floats( ));
+
+    lunchbox::ScopedWrite mutex( detail::_hdf5Lock );
+
+    try
+    {
+        detail::SilenceHDF5 silence;
+        H5::DataSet dataset = _file.openDataSet( _d_perimeters );
+
+        hsize_t dims;
+        const H5::DataSpace& dspace = dataset.getSpace();
+        if( dspace.getSimpleExtentNdims() != 1 ||
+            dspace.getSimpleExtentDims( &dims ) < 0 )
+        {
+            LBTHROW( std::runtime_error( "Reading morphology file '" +
+                          _file.getFileName() + "': bad number of dimensions in"
+                          " 'perimeters' dataspace"));
+        }
+
+        floatsPtr data( new floats( dims ));
+        dataset.read( data->data(), H5::PredType::NATIVE_FLOAT );
+        return data;
+    }
+    catch( ... )
+    {
+        if( getCellFamily() == FAMILY_GLIA )
+            LBTHROW( std::runtime_error( "No empty perimeters allowed for glia "
+                                         "morphology" ));
+        return floatsPtr( new floats( ));
```

Comment:
This line shouldn't be needed to avoid a warning if LBTHROW is declared as throwing. I don't know how to do that but _abort_ is doing the same kind of thing.